### ISSUE TITLE
More compact lsrules syntax

### ISSUE THREAD

**Issue by gemmmai** [naveednajam/Little-Snitch---Rule-Groups#8]
First I want to thank you for your great work! And I have another suggestion for optimisation:

Rule Group Subscriptions don't need creationDate, modificationDate, owner or process. It is possible thus to reduce every entry:

```
{
    "action" : "deny",
    "notes" : "Start lightswitch05",
    "remote-domains" : "digitalmedia.clickfunnels.com"
}
```

But in case of the Steven Black Hosts, where one just denies a lot of hosts, there is a new possibility since LS 4.2 to use an even more reduced syntax:

```
{
    "description" : " Date: 06 February 2019 15:27:38 (UTC) ,  Number of unique domains: 39,171 , ",
    "name" : "stevenblack.lsrules",
    "denied-remote-domains": [
"digitalmedia.clickfunnels.com", 
"drlivingood.clickfunnels.com", 
…
]
}
```

source: https://help.obdev.at/littlesnitch/ref-lsrules-file-format

**Comment by naveednajam** [comment]
In the beginning when I wrote this script I tried that idea however at that time it did not make any difference in performance and little snitch still used to hang with lots of domain, subdomains, and IP in a single rule. Creating separate rule is a better idea in my opinion as you can disable or enable rules selectively based on your requirement or temporarily without disabling the whole group.  I do agree removing creationDate, modificationDate, and owner will shed some bytes and make the file smaller. However I hardly believe that I will make any significant improvement to performance. but you can still take out these parameter by commenting out in the script. From my point of view at least having modified date is good for reference to see when was the last time these rules were updated and modified.

**Comment by gemmmai** [comment]
The proposed syntax changes would not have any significant effect on performance because LS only has to deal with the bloated code during import. A reduced code would positively impact traffic, if you have a lot of subscribers, that's all.

Keep up the good work!

**Comment by kaznovac** [comment]
I believe `notes` property can be dropped (this will potentially save one line per rule ~60k lines ~1.5mb).

this is a sample from these rules as exported by little snitch:

```
{
  "description" : "",
  "name" : "test",
  "rules" : [
    {
      "action" : "deny",
      "creationDate" : 1585039905.1522732,
      "modificationDate" : 1585039905.1522732,
      "owner" : "any",
      "process" : "any",
      "remote-domains" : "31.128.173.853.zhzy999.net31.128.173.853.zhzy999.net"
    },
    {
      "action" : "deny",
      "creationDate" : 1560426374.004019,
      "modificationDate" : 1560426374.004019,
      "owner" : "any",
      "process" : "any",
      "remote-domains" : "www.bar.ry2002.02-ry014.snpr.hotmx.hair.zaam.net  "
    },
...
```

and from little snitch lsrules reference https://help.obdev.at/littlesnitch/ref-lsrules-file-format `owner`, `creationDate` and `modificationDate` also seem optional.

**Comment by naveednajam** [comment]
Hi check out the [RuleGroupsV2](https://rulegroups.com). I have wrote the scripts from the ground up to make the rule groups more manageable and easier to subscribe. As many sources in stevenblack hosts repo are not updating I explored few more sources and also open to suggestion. 
New lists are using compact format however there are few limitation associated with compact format e.g we can not choose direction of traffic incoming/outgoing, unable to define priority are few limitation.
